lib/time.js

```diff
@@ -671,8 +671,10 @@ function CronTime(luxon) {
 
 		_hasAll: function (type) {
 			const constraints = CONSTRAINTS[TIME_UNITS.indexOf(type)];
+			const low = constraints[0];
+			const high = type === 'dayOfWeek' ? constraints[1] - 1 : constraints[1];
```

Comment:
It bothers me that 'dayOfWeek' is the only time unit being referenced as a plain string in the code, which was until now perfectly generic. But since we support it as an upper bound for cron string parsing, it still needs its constraints to be different than before.

Reply:
if you want you can do something like 
```
const dayOfWeekType = TIME_UNITS[5];
const high = type === dayOfWeekType ? constraints[1] - 1 : constraints[1];
```
to keep it generic

Reply:
@intcreator I expanded on that idea and found something that I think is even better, please let me know what you think:

```javascript
// top of the file
const TIME_UNITS_MAP = {
  SECOND: 'second',
  MINUTE: 'minute',
  HOUR: 'hour',
  DAY_OF_MONTH: 'dayOfMonth',
  MONTH: 'month',
  DAY_OF_WEEK: 'dayOfWeek'
};
const TIME_UNITS = Object.values(TIME_UNITS_MAP);
const TIME_UNITS_LEN = TIME_UNITS.length;

// inside `_hasAll()`
const high =
  type === TIME_UNITS_MAP.DAY_OF_WEEK
    ? constraints[1] - 1
    : constraints[1];
```